src/CodeGen_C.cpp

```diff
@@ -2024,7 +2024,13 @@ string CodeGen_C::print_assignment(Type t, const std::string &rhs) {
     if (cached == cache.end()) {
         id = unique_name('_');
         const char *const_flag = output_kind == CPlusPlusImplementation ? "const " : "";
-        stream << get_indent() << print_type(t, AppendSpace) << const_flag << id << " = " << rhs << ";\n";
+        if (t.is_handle()) {
+            // Don't print void *, which might lose useful type information. just use auto.
+            stream << get_indent() << "auto *";
```

Comment:
Looks like my suggestion about `auto *` was bad; I made a clone of this for testing and there are failures because we generated code of the form
```
pipeline_c.halide_generated.cpp:3720:17: error: unable to deduce ‘auto*’ from ‘0’
    auto *_275 = 0;
```

(see https://buildbot.halide-lang.org/master/#/builders/118/builds/541)

...that said, is just using `auto` here ok if the RHS is a plain `0`? If the type is Handle then surely we want to ensure it's at least pointer sized (if not explicitly 64 bits)...

Reply:
Ideally we could turn that 0 into 'nullptr'. I'll take a look.

Reply:
It was an unused value, so it was actually fine, but it was a place where we were explicitly deciding to use 0 for a pointer, so it was easy to change to nullptr

Reply:
now we get

```
pipeline_c.halide_generated.cpp:3720:17: error: unable to deduce ‘auto*’ from ‘nullptr’
    auto *_275 = nullptr;
```